### DISCUSSION TITLE
How to serialize custom type that appears in a nested dict rather than a model field

### DISCUSSION THREAD

**Discussion by DeanThompson** [pydantic/pydantic#8439] [Question]
I'm using Pydantic v2 with FastAPI. 

Model:

```
class MyModel(BaseModel):
    configuration: dict
```

The `configuration` is a dict that contains a lazy evaluated value by lazy_gettext. It may look like

```
from babel.support import LazyProxy

def gettext(message: str) -> str:
    return transtated_message

def lazy_gettext(message: str) -> LazyProxy
    return LazyProxy(gettext, message)
```

A `LazyProxy` is actually a string that only by evaluated until it's actually used like `str(lazy_value)`.

The `configuration` may look like

```
{
    "name": lazy_gettext("Field Name"),
   "other": {
        "foo": lazy_gettext("foo")
    }
}
```

When I'm trying to return MyModel from the FastAPI router, pydantic_core throws an exception

```
pydantic_core._pydantic_core.PydanticSerializationError: Unable to serialize unknown type: <class 'babel.support.LazyProxy'>
```

When using Pydantic v1, I'm able to add custom JSON encoder through the deprecated `json_encoders` like

```
class Config:
    json_encoders = {LocalProxy: str}
```

How can I solve this in Pydantic v2?

**Comment by pySilver** [comment]
@DeanThompson have you found solution?
